server/src/main/java/org/apache/gravitino/server/web/rest/PermissionOperations.java

```diff
@@ -87,12 +89,17 @@ public Response grantRolesToUser(
                       TreeLockUtils.doWithTreeLock(
                           NameIdentifier.of(AuthorizationUtils.ofRoleNamespace(metalake).levels()),
                           LockType.READ,
-                          () ->
-                              Utils.ok(
+                          () -> {
+                            try {
+                              return Utils.ok(
                                   new UserResponse(
                                       DTOConverters.toDTO(
                                           accessControlManager.grantRolesToUser(
-                                              metalake, request.getRoleNames(), user)))))));
+                                              metalake, request.getRoleNames(), user))));
+                            } catch (NoSuchRoleException nsr) {
```

Comment:
I think it is better to handle `IllegalRoleException` in the core, not convert to this exception in here.

Reply:
OK , just in my view, I usually throw `IllegalxxxException` in the other `Operation` class.  It's ok to throw in the core module.